interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -464,32 +492,32 @@ private[spel] class Typer(commonSupertypeFinder: CommonSupertypeFinder,
             reference.getName,
             typedParams.map(_.typingResult))
           ).map(TypingResultWithContext(_)) match {
-            case Right(x) => x.validNel
-            case Left(x) if strictMethodsChecking => x.invalidNel
-            case Left(_) => TypingResultWithContext(Unknown).validNel
+            case Right(x) => Writer(List.empty, x)
+            case Left(x) if strictMethodsChecking => Writer(List(x), TypingResultWithContext(Unknown))
+            case Left(_) => Writer(List.empty, TypingResultWithContext(Unknown))
           }
         }
       case Nil =>
-        InvalidMethodReference(reference.toStringAST).invalidNel
+        Writer(List(InvalidMethodReference(reference.toStringAST)), CollectedTypingResult.withEmptyIntermediateResults(TypingResultWithContext(Unknown)))
     }
   }
 
   @tailrec
   private def extractSingleProperty(e: PropertyOrFieldReference)
-                                   (t: SingleTypingResult): ValidatedNel[ExpressionParseError, TypingResult] = {
+                                   (t: SingleTypingResult): TypingR[TypingResult] = {
     t match {
       case typedObjectWithData: TypedObjectWithData =>
         extractSingleProperty(e)(typedObjectWithData.objType)
       case typedClass: TypedClass =>
         propertyTypeBasedOnMethod(typedClass, e)
           .orElse(MapLikePropertyTyper.mapLikeValueType(typedClass))
-          .map(Valid(_))
-          .getOrElse(NoPropertyError(t, e.getName).invalidNel)
+          .map(Writer(List.empty[ExpressionParseError], _))
+          .getOrElse(Writer(List(NoPropertyError(t, e.getName)), Unknown))
       case TypedObjectTypingResult(fields, objType, _) =>
         val typeBasedOnFields = fields.get(e.getName)
         typeBasedOnFields.orElse(propertyTypeBasedOnMethod(objType, e))
-          .map(Valid(_))
-          .getOrElse(NoPropertyError(t, e.getName).invalidNel)
+          .map(Writer(List.empty[ExpressionParseError], _))
```

Comment:
maybe we can extract method `def valid[T](result: T): Writer[...` for creating Writer with empty list? 
I think it would be easier to read this code then.

And maybe also `def invalid[T](errors: List[ExpressionParseError], result: T): Writer[..`

Reply:
Good point, added both. Also, I removed unnecessary wrapping with `TypingResultWithContext` which was done in many places but could be done in only a few once where it is imporant.